src/esp/gfx/RenderTarget.cpp

```diff
@@ -0,0 +1,296 @@
+        objectIdBuffer_{},
+        depthRenderTexture_{},
+        framebuffer_{Magnum::NoCreate},
+        depthUnprojection_{depthUnprojection},
+        depthShader_{depthShader},
+        unprojectedDepth_{NoCreate},
+        depthUnprojectionMesh_{NoCreate},
+        depthUnprojectionFrameBuffer_{NoCreate} {
+    if (depthShader_) {
+      CORRADE_INTERNAL_ASSERT(depthShader_->flags() &
+                              DepthShader::Flag::UnprojectExistingDepth);
+    }
+
+    colorBuffer_.setStorage(GL::RenderbufferFormat::SRGB8Alpha8, size);
+    objectIdBuffer_.setStorage(GL::RenderbufferFormat::R32UI, size);
+    depthRenderTexture_.setMinificationFilter(GL::SamplerFilter::Nearest)
+        .setMagnificationFilter(GL::SamplerFilter::Nearest)
+        .setWrapping(GL::SamplerWrapping::ClampToEdge)
+        .setStorage(1, GL::TextureFormat::DepthComponent32F, size);
+
+    framebuffer_ = GL::Framebuffer{{{}, size}};
+    framebuffer_.attachRenderbuffer(RgbaBuffer, colorBuffer_)
+        .attachRenderbuffer(ObjectIdBuffer, objectIdBuffer_)
+        .attachTexture(GL::Framebuffer::BufferAttachment::Depth,
+                       depthRenderTexture_, 0)
+        .mapForDraw({{0, RgbaBuffer}, {1, ObjectIdBuffer}});
+    CORRADE_INTERNAL_ASSERT(
+        framebuffer_.checkStatus(GL::FramebufferTarget::Draw) ==
+        GL::Framebuffer::Status::Complete);
+
+#ifndef CORRADE_TARGET_EMSCRIPTEN
+    GL::Renderer::setClearColor(Magnum::Color4{0, 0, 0, 1});
+    GL::Renderer::setClearDepth(1.0);
+#endif
+  }
+
+  void initDepthUnprojector() {
+    if (depthUnprojectionMesh_.id() == 0) {
+      unprojectedDepth_ = GL::Renderbuffer{};
+      unprojectedDepth_.setStorage(GL::RenderbufferFormat::R32F,
+                                   framebufferSize());
+
+      depthUnprojectionFrameBuffer_ = GL::Framebuffer{{{}, framebufferSize()}};
+      depthUnprojectionFrameBuffer_
+          .attachRenderbuffer(UnprojectedDepthBuffer, unprojectedDepth_)
+          .mapForDraw({{0, UnprojectedDepthBuffer}});
+      CORRADE_INTERNAL_ASSERT(
+          framebuffer_.checkStatus(GL::FramebufferTarget::Draw) ==
+          GL::Framebuffer::Status::Complete);
+
+      depthUnprojectionMesh_ = GL::Mesh{};
+      depthUnprojectionMesh_.setCount(3);
+    }
+  }
+
+  void unprojectDepthGPU() {
+    CORRADE_INTERNAL_ASSERT(depthShader_ != nullptr);
+    initDepthUnprojector();
+
+    depthUnprojectionFrameBuffer_.bind();
+    depthShader_->bindDepthTexture(depthRenderTexture_)
+        .setDepthUnprojection(depthUnprojection_);
+
+    depthUnprojectionMesh_.draw(*depthShader_);
+  }
+
+  void renderEnter() {
+#ifdef CORRADE_TARGET_EMSCRIPTEN
+    framebuffer_.clearDepth(1.0);
+    framebuffer_.clearColor(0, Magnum::Color4{});
+    framebuffer_.clearColor(1, Magnum::Vector4ui{});
+#else
+    framebuffer_.clear(GL::FramebufferClear::Color |
+                       GL::FramebufferClear::Depth);
+    framebuffer_.clearColor(1, Magnum::Vector4ui{});
+#endif
```

Comment:
Forgive me if I forgot already, but what was the reason for doing this? Is the second variant faster on desktop?